### PR TITLE
feat: `is_mutating` method

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

Part of the work on [SIP-117](https://github.com/apache/superset/issues/26786), this PR introduces a `has_mutation()` method to `SQLScript`, as well as a `is_mutating()` method to `SQLStatement`, in order to check if the SQL contains any DML.

This also fixes a bug where this SQL is flagged incorrectly as DML:

```sql
with source as (
  select 1 as one
)
select * from source
```

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TESTING INSTRUCTIONS

Added tests.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/sql_parse.py [hunk 1]
```diff
@@ -522,6 +530,43 @@ def _extract_tables_from_statement(
...
+        """
+        for node in self._parsed.walk():
+            if isinstance(
+                node,
+                (
+                    exp.Insert,
+                    exp.Update,
+                    exp.Delete,
+                    exp.Merge,
+                    exp.Create,
```
Comment (mistercrunch) [inline superset/sql_parse.py:547]: technically this is DDL not DML, wondering if `is_mutating` would be a better name?

**Review by mistercrunch — APPROVED** [review]
LGTM outside of the naming concern, pre-approving assuming the name change

File: superset/sql_parse.py (hunk 1 above)
Comment (betodealmeida) [reply superset/sql_parse.py:547]: @mistercrunch do you think we should also rename the DB modal? It currently says "Allow DML".

Comment (mistercrunch) [reply superset/sql_parse.py:547]: yes I would. Here's what it says currently ->  Screenshot 2024-09-09 at 9 11 58 AM

`Allow DML and DDL` - could be good to unpack the acronyms in the tooltip too. "Allow the execution of DDL (Data Definition Language, ie CREATE, DROP, TRUNCATE, ...) and DML (Data Modification Language, ie INSERT, UPDATE, DELETE, ...) {...}"

**Pull request merged**